daemon/main.cpp

```diff
@@ -1884,10 +1884,15 @@ int Daemon::answer_client_requests()
             }
 
             for (map<string, NativeEnvironment>::iterator it = native_environments.begin();
-                 it != native_environments.end(); ++it) {
+                 it != native_environments.end(); ) {
                 if (it->second.create_env_pipe && FD_ISSET(it->second.create_env_pipe, &listen_set)) {
-                    create_env_finished(it->first);
+                    if(!create_env_finished(it->first))
+                    {
+                        native_environments.erase(it++);
+                        continue;
+                    }
                 }
+                ++it;
```

Comment:
No...this is a pretty standard loop that erases elements based on a condition. It looks fine to me.